test/process.js

```diff
@@ -47,6 +47,10 @@ test('process(file, done)', () => {
     })
     .use(
       () =>
+        /**
+         * @param {Node} tree
+         * @param {VFile} file
+         */
```

Comment:
these are needed by type coverage, `tree` was considered an implicit `any`

Reply:
Was TypeScript able to infer its type before? This seems like a regression in TypeScript.

Reply:
Good question, inside VSCode, both TS4.9 and TS5.0 show `Node<Data>` even without this annotation.
`type-coverage` is where the error appears

```
unified/test/process.js:50:19: tree
unified/test/process.js:103:19: tree
unified/test/run.js:552:19: tree
unified/test/run.js:562:19: _
unified/test/use.js:255:19: node
2855 / 2860 99.82%
```

Reply:
It is version v2.25.0 ✅

Reply:
Gah, I didn't have VSCode manually configured to ts5.0.2
TypeScript itself does throw
```
Parameter 'tree' implicitly has an 'any' type.ts(7006)
```

this may be a TS regression.

Reply:
They’re asking the community to report regressions. I think this one would be a most welcome issue upstream. :)

I would like to update to TypeScript 5, but unless we want to use any new TypeScript features, I’m not in a hurry to update. I’d rather wait for an upstream fix than adding workarounds.

Reply:
Hmm, this is an interesting one, when I try to replicate the issue in the TS Playground, it works again.
[playground](https://www.typescriptlang.org/play?target=99&moduleResolution=99&module=100&ts=5.0.2&noUncheckedIndexedAccess=true&noUnusedParameters=true&esModuleInterop=false&declaration=false&jsx=0&exactOptionalPropertyTypes=true&noFallthroughCasesInSwitch=true&noImplicitOverride=true&noPropertyAccessFromIndexSignature=true&useUnknownInCatchVariables=true&noUnusedLocals=true#code/JYOwLgpgTgZghgYwgAgAoHtRmQbwL4BQBEAHgA7pTZbTxJroDOwYw6IuByyjYcVALgZYA3F2QQQAEyEZR40FMlgA-EJABXALYAjaAG0AusgA+yDdIgxQEKWMLFylauFqIUAOXRKAPABUAETg+CRJIaUZkdB0AKwgEbABeZCC+AD5ObjAATzIIIV4oUABzMW4pYLg1ZEDK03NLaxBbMuQKZlZ2aowOtg4zCyUmloIHUgoqZBpYdxS6nHF9AGsIbIKwIpBiwyELJZB0AHcQeyIcvOQAVUYIVAAbDWLQH3FUfhu-KAgUUnCpSK8SnqADdMFJkMlQcApAAacQAYQ0UC+4E+31CfwB3hQZih4MhYLh3Hh6C0ZGAdwgaJ+YUk-2QgJxyDxEOZhIRpPJlIAShBGBo7kk2dCicgAJIgMgaIV40UAeWlUplYIIGWSCrASoxdKxSnEKnFkul2oiPA2JXE3ANqCg6CQjEYlBe3BdLo1StFru4AGU8ghgNYED53dKYchEcjlNS0p6vb74gHgEGQ2AwySyRSqV8IDHLV701yILz+YK89w0nmhBKtb8dQzsXnrbb7Y6oM6vT6-Ymg9XQ2h3lnvrmO9wU7HXfH-YGfAXM9SwynhyPZzy+QKwGXkBXXeoIMDoOIq0bsLXTYz9Whm3zW+2J13p73U-2oB9s0u7wmH8e00iUWBo+OhoepuKZ5tuyCyFeDpOm8L6DhAP6Rqi2Zppyc4oeGaGriWYBpGIBDnCgk7dj4AAyVjKiKyDcsAxQABaUVIarUbRDEmvSLIGuRMDYEINH0WA+HTHQKAAGoAGKZrgYxOJMhE1FAcAgIwMCUFo8JwHcdw6IgSz+Nm7G6igySMsxAAU4jQLaUDVAAosilD1JoWn1IMVg2LC4gHEo1TUq5jQeaK1iUtUElSQMAXNFIBAAJQQhkeL4eMzjIPJnxKSpanQO2j6GfWQImdi8qKsap70oyrKPqqrIWdw3n5EBobiMFDVhZSorNGEQjpcpqlQOpmnabpwYlbhsXxZepLADcI2asaEVDB5IJghkZgpvU9nWf5i1Rct0JJbJ2DyfcjzPK8DxPCAsFwFoECQC+eVKdkRisk9RiirlhVKMVc1ClV5niGAdHTUeNa0meDauga61lUZm5Nna15OrKjVjR2kGI9BbYo4+Ya48gi6VgTo15eeUOTS2TpjsTv3gdwGOU1AooAHSszcYCsFsjCyBdoDXbd92MONiQZLlsP5RAF4w+D5WQy6Bo9Zl-XZfji57dFO7ChrEFa-hx1QTerwDn54sVbiYKsrKCK-lGBmm9i6uW+yxJYfBpMO+b0JOyKHIZth67e9FzELNwZkxQzSNtrBr7fIhf7zphfvwahSfFuueFENwGgzXmJ2Xfzd3QJE4tvcYySl7Gn3IMR06MnHttDrG63JFV5a1S6ZC8yAPOnSAPh53z-A3YXL549+NNKu+yCs8z7Oc8U3NoF3BeC-U+g6Og6CUkphjiOHVw3APfe58bKF5hG8dn66K7J+frtp4KlffnmoHlvYB0TEduQoLZACOGiaUYD4AAGmGAAmuZfwaQw4TT8HlYByADQAEYdYACY4ri3bH4VUMCRY1DymAxByAUFCFQReDYGhJb02QPAO4NwxDyVsuQeIYB-B5QoTmVkfh8IIHYLwBoiZbCsnwMgOAkQbSY1bGICwgipBh2ZtnCA7dcF0xoRYBIfRkBmQ2LHGhmY4ohy9GlPkQomF+lYX-ABdCfCEXQDAVKGFTIZw7MY3gqDWRmJYT4SxgCbHfzsXo9qyA2o5mcS6QgMUxBAA)

Reply:
> I would like to update to TypeScript 5, but unless we want to use any new TypeScript features, I’m not in a hurry to update. I’d rather wait for an upstream fix than adding workarounds.

💯 agree

Reply:
Update, as of TypeScript 5.1.6 and Type Coverage 2.26.0, it is still an `any` type.
`typescript` does not throw anymore, but `type-coverage` does.

```
unified/test/process.js:50:19: tree
2865 / 2866 99.96%
The type coverage rate(99.96%) is lower than the target(100%).
```